nornir/plugins/inventory/netbox.py

```diff
@@ -86,14 +88,33 @@ def __init__(
                 host["data"]["role"] = d["device_role"]["slug"]
                 host["data"]["model"] = d["device_type"]["slug"]
 
-                # Attempt to add 'platform' based of value in 'slug'
-                host["platform"] = d["platform"]["slug"] if d["platform"] else None
+                if use_platform_args:
```

Comment:
Not sure why you are changing this logic here. Platform for a host should always be set, independent of `use_platform_args`